tests/unit/utils/test_utils.py

```diff
@@ -878,3 +878,31 @@ def test_parse_duration_hypothesis(duration):
 ])
 def test_mimetype_extension(mimetype, extension):
     assert utils.mimetype_extension(mimetype) == extension
+
+
+class TestCleanupFileContext:
+
+    def test_no_file(self, tmpdir):
+        tmpfile = tmpdir / 'tmp.txt'
+        with utils.cleanup_file(tmpfile):
+            pass
+        assert not os.path.exists(tmpfile)
+
+    def test_no_error(self, tmpdir):
+        tmpfile = tmpdir / 'tmp.txt'
+        with tmpfile.open('w'):
+            pass
+        with utils.cleanup_file(tmpfile):
+            pass
+        assert not os.path.exists(tmpfile)
+
+    def test_error(self, tmpdir):
+        tmpfile = tmpdir / 'tmp.txt'
+        with tmpfile.open('w'):
+            pass
+        try:
```

Comment:
Using `with pytest.raises(RuntimeError):` here instead would make sure that the exception is also correctly bubbled through - see https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions

Reply:
I had that at first but my linter then complains that the last line is unreachable. But I didn't actually try if the linter used in the CI also thinks that.

Reply:
Did you have the last line outside of the `with`-block? If it was inside, that would've been correct - the code after the exception isn't run.

Reply:
Yes, outside the `with`-block. Seems to maybe be a bug with my linter indeed (`coc-pyright`). Anyway, changed to use `pytest.raises`